src/Hl7.Fhir.Specification/Validation/TypeRefValidationExtensions.cs

```diff
@@ -102,14 +101,30 @@ OperationOutcome validate()
                 var validations = tr.GetDeclaredProfiles()?.Select(profile => createValidatorForDeclaredProfile(validator, instance, profile));
                 var result = validator.Combine(BatchValidationMode.Any, instance, validations);
 
-                // If this is a reference, also validate the reference against the targetProfile
-                if (ModelInfo.FhirTypeNameToFhirType(tr.Code) == FHIRAllTypes.Reference)
-                    result.Add( validator.ValidateResourceReference(instance, tr) );
+                result.Add(validateTargetProfiles(validator, instance, tr));
 
                 return result;
             }
         }
 
+        internal static OperationOutcome ValidateTargetProfiles(this Validator validator, ElementDefinition definition, ScopedNode instance)
```

Comment:
This one file now has a method `ValidateTargetProfiles()` and a method `validateTargetProfiles()` with almost the same signature.

Reply:
I think the method starting with lowercase is the original behaviour, so it is actually not validating the target profile?

Reply:
In the method `ValidateChildConstraints` I wanted only to validate the targetprofiles, which is part of the method `ValidateTypeReferences`. To not duplicate code I wrote a new method `ValidateTargetProfiles` which is used by the ChildConstraints and `ValidateTypeReferences`.
I have now decided to use the method `ValidateTypeReferences` in `ValidateChildConstraints`. That does also validate the targetprofiles and the declared profiles. The issue was only about targetprofiles, but to validate the declared profiles as well does not hurt.